mailbox/store/src/main/java/org/apache/james/mailbox/store/search/MessageSearches.java

```diff
@@ -251,7 +252,7 @@ private boolean textContains(String value, MailboxMessage message) throws IOExce
     }
 
     private InputStream textHeaders(MailboxMessage message) throws MimeIOException, IOException {
-        ByteArrayOutputStream out = new ByteArrayOutputStream();
+        ByteArrayOutputStream out = new ByteArrayOutputStream(Ints.checkedCast(message.getFullContentOctets()));
```

Comment:
Yes, ok, we can introduce a *super class* of ByteArrayInputStream in James-utils that gain access to underlying byte array (and that don't fail on negative sizes).

Is it worth investigating this?